modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -5288,11 +5277,51 @@ void GDScriptAnalyzer::downgrade_node_type_source(GDScriptParser::Node *p_node)
 }
 
 void GDScriptAnalyzer::mark_lambda_use_self() {
-	for (GDScriptParser::LambdaNode *lambda : lambda_stack) {
+	GDScriptParser::LambdaNode *lambda = current_lambda;
+	while (lambda != nullptr) {
 		lambda->use_self = true;
+		lambda = lambda->parent_lambda;
 	}
 }
 
+void GDScriptAnalyzer::resolve_pending_lambda_bodies() {
```

Comment:
```suggestion
void GDScriptAnalyzer::resolve_pending_lambda_bodies() {
	if (pending_body_resolution_lambdas.is_empty()) {
		return;
	}

```

Reply:
Do you want to do anything with this? It looks sensible.

Reply:
Did you test did, Danil?

Reply:
I mean, it just avoids moving references around and clearing an already empty container.